### PR TITLE
adding caikit-style prompt adapter support

### PR DESCRIPTION
**Pull request by SwapnilDreams100**
Hey @joerunde, solving https://github.ibm.com/ai-foundation/fmaas-inference-server/issues/732 by extending the PromptAdapterModel to load caikit style adapters. 
Tested using the bloom adapter here: https://github.com/IBM/text-generation-inference/tree/main/integration_tests/prompt_prefixes/bloom_sentiment_1

### CODE REVIEW

File: vllm/tgis_utils/prompt_adapter.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    @classmethod
+    def from_local_checkpoint(cls,
+                              adapter_path,
+                              prompt_adapter_id,
+                              torch_device='cuda') -> "PromptAdapterModel":
+        try:
+            adapters_weights = load_peft_weights(adapter_path,
+                                                torch_device)
+            prompt_embedding = adapters_weights["prompt_embeddings"].half()
+        except Exception as e:
```
Comment (njhill) [inline vllm/tgis_utils/prompt_adapter.py:26]: Is there a cleaner way to check whether this is a PEFT type adapter?

Like: if `adapter_config.json` is found, elif `decoder.pt` is found else raise "no supported adapter format found at path {path}"   .. kind of thing

**Review by njhill — COMMENTED** [review]
Thanks @SwapnilDreams100 this looks great.

As well as inline comment, I think we'll want to monkey-patch this into vLLM if poss from our tgis adapter layer rather than modifying the models.py file.

**Comment by njhill** [comment]
@SwapnilDreams100 also you need to sign-off your commits.

**Comment by prashantgupta24** [comment]
closing in favor of https://github.com/IBM/vllm/pull/46

**Pull request closed without merging**